datalad/support/annexrepo.py

```diff
@@ -699,6 +699,16 @@ def is_remote_annex_ignored(self, remote):
             default=False
         )
 
+    def set_remote_ignore(self, remote, ignored=False):
+        """Add or set an annex-ignore section for a given remote"""
+        # check if there is a section already
+        if self.config.has_section('remote.{}.annex-ignore'.format(remote)):
```

Comment:
well, indeed it returns True

```shell
$> python3 -c 'import datalad.api as dl; print(dl.Dataset(".").config.has_section("remote.origin.annex-ignore"))'  
True
```
which somewhat surprises me since this setting is not an entire section but just one option, so wouldn't a more semantically correct :?

```shell
$> python3 -c 'import datalad.api as dl; print(dl.Dataset(".").config.has_option("remote.origin", "annex-ignore"))' 
True
```